### ISSUE TITLE
Unable to uninstall single-node k3s cluster

### ISSUE THREAD

**Issue by pixiake** [kubesphere/kubekey#1148]
### What is version of KubeKey has the issue?

v2.0.0

### What is your os environment?

Ubuntu20.04

### A clear and concise description of what happend.

I installed k3s cluster with `./kk create cluster --with-kubernetes v1.21.6-k3s`. But if I don't specify the configuration file, I can't uninstall k3s cluster with `./kk delete cluster`.

I think we can check whether `k3s-uninstall.sh` exists in the delete cluster pipeline, and if so, uninstall k3s first.

**Comment by xiaods** [comment]
This is designed to be like this, not a bug. we need discussion here and give a good choice then do action.

please read the file: 
# /kubekey/pkg/pipelines/delete_cluster.go
```
func DeleteCluster(args common.Argument) error {
	var loaderType string
	if args.FilePath != "" {
		loaderType = common.File
	} else {
		loaderType = common.AllInOne
	}

	runtime, err := common.NewKubeRuntime(loaderType, args)
	if err != nil {
		return err
	}

	switch runtime.Cluster.Kubernetes.Type {
	case common.K3s:  <--- follow this logic, runtime read kubeconfig and get runtime, it will know the type is k3s.
		if err := NewK3sDeleteClusterPipeline(runtime); err != nil {
			return err
		}
	case common.Kubernetes:
		if err := NewDeleteClusterPipeline(runtime); err != nil {
			return err
		}
	default:  
		if err := NewDeleteClusterPipeline(runtime); err != nil {
			return err
		}
	}
	return nil
}
```

are you sure uninstall k3s failed?

**Comment by xiaods** [comment]
@pixiake any update on this issue.

**Comment by pixiake** [comment]
@xiaods 
There should be no problem if we use the configuration file，because the `Kubernetes.Type` is specified in the configuration file.

If we use `allinone` mode, kk will automatically generate a default configuration, in which no  `Kubernetes.Type`  is specified.

**Comment by xiaods** [comment]
> with-kubernetes

@pixiake  cloud you please give a Kubernetes.Type sample case in your configfile. i read the config template, don't found any type for specified the k8s type. Maybe I have not totally understand your case. please let me know.

/kubekey/config/templates/cluster.go  --- this is default kk config.
```
// Cluster defines the template of cluster configuration file default.
var Cluster = template.Must(template.New("Cluster").Parse(
	dedent.Dedent(`
apiVersion: kubekey.kubesphere.io/v1alpha2
kind: Cluster
metadata:
  name: {{ .Options.Name }}
spec:
  hosts:
  - {name: node1, address: 172.16.0.2, internalAddress: 172.16.0.2, user: ubuntu, password: "Qcloud@123"}
  - {name: node2, address: 172.16.0.3, internalAddress: 172.16.0.3, user: ubuntu, password: "Qcloud@123"}
  roleGroups:
    etcd:
    - node1
    control-plane: 
    - node1
    worker:
    - node1
    - node2
  controlPlaneEndpoint:
    ## Internal loadbalancer for apiservers 
    # internalLoadbalancer: haproxy

    domain: lb.kubesphere.local
    address: ""
    port: 6443
  kubernetes:
    version: {{ .Options.KubeVersion }}
    clusterName: cluster.local
    autoRenewCerts: true
    containerManager: {{ .Options.ContainerManager }}
  etcd:
    type: kubekey
  network:
    plugin: calico
    kubePodsCIDR: 10.233.64.0/18
    kubeServiceCIDR: 10.233.0.0/18
    ## multus support. https://github.com/k8snetworkplumbingwg/multus-cni
    multusCNI:
      enabled: false
  registry:
    privateRegistry: ""
    namespaceOverride: ""
    registryMirrors: []
    insecureRegistries: []
  addons: []

{{ if .Options.KubeSphereEnabled }}
{{ .Options.KubeSphereConfigMap }}
{{ end }}
    `)))
```

**Comment by xiaods** [comment]
I have reproduce this bug on aws clean server, yes, the delete cluster step only perform kubeadm delete steps not k3s-uninstall.sh.

```
[root@ip-172-31-27-251 ~]# kk delete cluster


 _   __      _          _   __           
| | / /     | |        | | / /           
| |/ / _   _| |__   ___| |/ /  ___ _   _ 
|    \| | | | '_ \ / _ \    \ / _ \ | | |
| |\  \ |_| | |_) |  __/ |\  \  __/ |_| |
\_| \_/\__,_|_.__/ \___\_| \_/\___|\__, |
                                    __/ |
                                   |___/

01:00:21 UTC [GreetingsModule] Greetings
01:00:21 UTC message: [ip-172-31-27-251.ap-northeast-2.compute.internal]
Greetings, KubeKey!
01:00:21 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:21 UTC [DeleteClusterConfirmModule] Display confirmation form
Are you sure to delete this cluster? [yes/no]: yes
01:00:24 UTC success: [LocalHost]
01:00:24 UTC [ResetClusterModule] Reset the cluster using kubeadm
01:00:24 UTC stdout: [ip-172-31-27-251.ap-northeast-2.compute.internal]
/bin/bash: /usr/local/bin/kubeadm: No such file or directory
01:00:24 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:24 UTC [ClearOSModule] Reset os network config
01:00:24 UTC stdout: [ip-172-31-27-251.ap-northeast-2.compute.internal]
/bin/bash: ipvsadm: command not found
01:00:24 UTC stdout: [ip-172-31-27-251.ap-northeast-2.compute.internal]
Cannot find device "nodelocaldns"
01:00:24 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:24 UTC [ClearOSModule] Uninstall etcd
01:00:24 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:24 UTC [ClearOSModule] Remove cluster files
01:00:24 UTC stdout: [ip-172-31-27-251.ap-northeast-2.compute.internal]
rm: cannot remove ‘/var/lib/kubelet/pods/1dd82477-f7f7-414e-ad81-34f5863e03aa/volumes/kubernetes.io~projected/kube-api-access-6gp2k’: Device or resource busy
rm: cannot remove ‘/var/lib/kubelet/pods/798ba461-2142-4bb8-8513-69b76520b0f3/volumes/kubernetes.io~projected/kube-api-access-wcqbb’: Device or resource busy
rm: cannot remove ‘/var/lib/kubelet/pods/2360d265-3a18-4391-a4a9-b46c941b13bd/volumes/kubernetes.io~projected/kube-api-access-fzgzf’: Device or resource busy
01:00:25 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:25 UTC [ClearOSModule] Systemd daemon reload
01:00:25 UTC success: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:25 UTC [UninstallAutoRenewCertsModule] UnInstall auto renew control-plane certs
01:00:25 UTC skipped: [ip-172-31-27-251.ap-northeast-2.compute.internal]
01:00:25 UTC Pipeline[DeleteClusterPipeline] execute successfully

```

**Comment by xiaods** [comment]
I add log in deletecluster func,
```
[root@ip-172-31-27-251 amd64]# ./kk delete cluster
v1.23.9
============================================================
kubernetes
```
the kubekey init runtime on mem, it dont read any k8s version from anywhere. so it give a default k8s version and type on runtime. it redirect wrong path to exec delete cluster pipeline.

**Comment by xiaods** [comment]
for consistent design, the kk delete should add --with-kubernetes args to support specifed k8s version. it will resolve this concerns.

**Comment by xiaods** [comment]
@pixiake  hope my pr can fix your case.

**Comment by 24sama** [comment]
Fixed by: #1426
